doc/source/security/innodb-encryption.rst

```diff
@@ -0,0 +1,405 @@
+
+If the requested ``percona-innodb-`` does not exist in the keyring, |Percona Server| creates the key with version 1. If a ``percona-innodb-`` key cannot be created with the requested ID, the ``CREATE TABLE`` statement fails.
+
+.. seealso::
+
+    MySQL Documentation: ``default_table_encryption``
+    https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_default_table_encryption
+
+.. variable:: table_encryption_privilege_check
+
+  :cli: table-encryption-privilege-check
+  :implemented: version 8.0.16-7
+  :dyn: Yes
+  :scope: Global
+  :vartype: Boolean
+  :default: OFF
+
+The variable is used when creating or altering a general tablespace or table with an encryption setting different than the :variable:`default_table_encryption`. The default value is `OFF`.
+
+.. seealso::
+
+  MySQL Documentation: ``table_encryption_privilege_check``
+  https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html
+
+.. variable:: innodb_encryption_threads
+
+   :cli: ``--innodb-encryption-threads``
+   :dyn: Yes
+   :scope: Global
+   :vartype: Numeric
+   :default: 0
+
+This variable works in combination with the :variable:`default_table_encryption`
+variable set to ``ONLINE_TO_KEYRING``. This variable configures the number of
+threads for background encryption. For the online encryption to work, this
+variable must contain a value greater than **zero**.
+
+.. variable:: innodb_online_encryption_rotate_key_age
+
+   :cli: ``--innodb-online-encryption-rotate-key-age``
+   :dyn: Yes
+   :scope: Global
+   :vartype: Numeric
+   :default: 1
+
+By using this variable, you can re-encrypt the table encrypted using
+KEYRING. The value of this variable determines how frequently the encrypted
+tables should be encrypted again. If it is set to **1**, the encrypted table is
+re-encrypted on each key rotation. If it is set to **2**, the table is encrypted
+on every other key rotation.
+
+.. variable:: innodb_encrypt_online_alter_logs
+
+   :cli: ``--innodb-encrypt-online-alter-logs``
+   :dyn: Yes
+   :scope: Global
+   :vartype: Boolean
+   :default: OFF
+
+This variable simultaneously turns on the encryption of files used by InnoDB for
+full text search using parallel sorting, building indexes using merge sort, and
+online DDL logs created by InnoDB for online DDL. Encryption is available for file merges used in queries and backend processes.
+
+Undo Tablespace Encryption
+==============================================================================
+
+Implemented in :rn:`8.0.16-7`, the undo tablespace data encryption is available as an option. The undo data encryption must be enabled; the feature is disabled by default. When the undo log data is written to disk, the data is encrypted using the tablespace encryption key. The undo log data is decrypted when read.
```

Comment:
upstream feature, and it started with an earlier version. .13 or .11?

"The undo data encryption must be enabled; the feature is disabled by default."?

Reply:
The document was removed and the text added to the appropriate docs. 
The undo data sentence was added to encrypting undo tablespaces